src/main/kotlin/io/gitlab/arturbosch/detekt/sonar/sensor/DetektorConfiguration.kt

```diff
@@ -16,55 +16,76 @@ import io.gitlab.arturbosch.detekt.sonar.foundation.NoAutoCorrectConfig
 import org.sonar.api.config.Settings
 import java.io.File
 
 /**
  * @author Artur Bosch
  */
-fun configureDetektor(context: SensorContext): Detektor {
-	val fileSystem = context.fileSystem()
-	val baseDir = fileSystem.baseDir()
-	val settings = context.settings()
+class DetektorConfiguration(context: SensorContext) {
 
-	val pathFiltersString = settings.getString(PATH_FILTERS_KEY) ?: PATH_FILTERS_DEFAULTS
-	val filters = pathFiltersString.split(SEPARATOR_SEMICOLON, SEPARATOR_COMMA).map { PathFilter(it) }
+	val config: Config
+	val settings: Settings
+	val baseDir: File
 
-	val config = chooseConfig(baseDir, settings)
-	val processingSettings = ProcessingSettings(
-			baseDir.toPath(), NoAutoCorrectConfig(config), filters)
+	init {
+		val fileSystem = context.fileSystem()
+		baseDir = fileSystem.baseDir()
+		settings = context.settings()
+		config = chooseConfig(baseDir, settings)
+	}
 
-	return DetektFacade.instance(processingSettings)
-}
+	/**
+	 * Provides an instance of [Detektor] to be used for whole project analysis.
+	 */
+	fun configureDetektor(): Detektor {
+		val pathFiltersString = settings.getString(PATH_FILTERS_KEY) ?: PATH_FILTERS_DEFAULTS
+		val filters = pathFiltersString.split(SEPARATOR_SEMICOLON, SEPARATOR_COMMA).map { PathFilter(it) }
+		val processingSettings = ProcessingSettings(baseDir.toPath(), NoAutoCorrectConfig(config), filters)
 
-private fun chooseConfig(baseDir: File, settings: Settings): Config {
-	val configPath = settings.getString(CONFIG_PATH_KEY)
-	val externalConfigPath = configPath?.let {
-		LOG.info("Registered config path: $it")
-		val configFile = File(it)
-		if (!configFile.isAbsolute) { // TODO find out how to resolve always to root path, not module path
-			val resolved = baseDir.resolve(configPath)
-			LOG.info("Relative path detected. Resolving to project dir: $resolved")
-			resolved
-		} else configFile
+		return DetektFacade.instance(processingSettings)
 	}
 
-	val internalConfigResource = settings.getString(CONFIG_RESOURCE_KEY)
-			?.let { if (it.isNullOrBlank()) null else it }
+	/**
+	 * Provides an instance of [Detektor] for a specific [inputFile]
+	 */
+	fun configureDetektor(inputFile: InputFile): Detektor {
+		val processingSettings = ProcessingSettings(inputFile.path(), NoAutoCorrectConfig(config), emptyList())
 
-	val possibleParseArguments = Args().apply {
-		config = externalConfigPath?.path
-		configResource = internalConfigResource
+		return DetektFacade.instance(processingSettings)
 	}
 
-	val bestConfigMatch = possibleParseArguments.loadConfiguration()
-	val bestConfigMatchOrDefault = bestConfigMatch.let {
-		if (it == Config.empty) {
-			LOG.info("No detekt yaml configuration file found, using the default configuration.")
-			DEFAULT_YAML_CONFIG
-		} else it
+	private fun chooseConfig(baseDir: File, settings: Settings): Config {
+		val configPath = settings.getString(CONFIG_PATH_KEY)
+		val externalConfigPath = configPath?.let {
+			LOG.info("Registered config path: $it")
```

Comment:
About code agreements I tend to define name for `it` if the block after is long and it is used in multiple places